unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1092,7 +1092,7 @@ public virtual ObjectMetadata[] generateObjectMetadata()
             SimObjPhysics[] simObjects = null;
             if (this.simObjFilter != null) {
                 foreach (SimObjPhysics sop in this.simObjFilter) {
-                    if (isSimObjVisible(m_Camera, sop)) {
+                    if (isSimObjVisible(m_Camera, sop, false, this.maxVisibleDistance)) {
```

Comment:
This sort of fixes the problem but not really as the `VisibleSimObjs` can result in an object being invisible using the filtered version and visible with the non-filtered version. I vote for adding a `simObjFilter` parameter to `VisibleSimObjs` and then having the `VisibleSimObjs` logic skip anything not in this filter. At the cost of some computation we get guaranteed consistency.

Reply:
I'm not clear how that could occur - the only difference between VisibleSimObjs and isSimObjVisible is the former performs a capsule cast to filter down the list of candidates that we then send to isSimObjVisible with the same distance value as the filtered list.